tests/micro/zephyr/test_zephyr_aot.py

```diff
@@ -0,0 +1,217 @@
+    It is used to capture the tensor data (for both inputs and expected outputs).
+    """
+    file_path = pathlib.Path(f"{output_path}/" + tensor_name).resolve()
+    # create header file
+    raw_path = file_path.with_suffix(".h").resolve()
+    with open(raw_path, "w") as header_file:
+        header_file.write("#include <stddef.h>\n")
+        header_file.write("#include <stdint.h>\n")
+        header_file.write("#include <dlpack/dlpack.h>\n")
+        header_file.write(f"const size_t {tensor_name}_len = {npy_data.size};\n")
+
+        if npy_data.dtype == "int8":
+            header_file.write(f"int8_t {tensor_name}[] =")
+        elif npy_data.dtype == "int32":
+            header_file.write(f"int32_t {tensor_name}[] = ")
+        elif npy_data.dtype == "uint8":
+            header_file.write(f"uint8_t {tensor_name}[] = ")
+        elif npy_data.dtype == "float32":
+            header_file.write(f"float {tensor_name}[] = ")
+
+        header_file.write("{")
+        for i in np.ndindex(npy_data.shape):
+            header_file.write(f"{npy_data[i]}, ")
+        header_file.write("};\n\n")
+
+
+def _read_line(fd):
+    data = ""
+    new_line = False
+    while True:
+        if new_line:
+            break
+        new_data = fd.read(1, timeout_sec=10)
+        for item in new_data:
+            new_c = chr(item)
+            data = data + new_c
+            if new_c == "\n":
+                new_line = True
+                break
+    return data
+
+
+def _get_result_line(fd):
+    while True:
+        data = _read_line(fd)
+        if "result" in data:
+            return data
+
+
+def test_tflite(platform, west_cmd):
+    """Testing a TFLite model."""
+    model, zephyr_board = PLATFORMS[platform]
+    input_shape = (1, 32, 32, 3)
+    output_shape = (1, 10)
+
+    this_dir = os.path.dirname(__file__)
+    tvm_source_dir = os.path.join(this_dir, "..", "..", "..")
+    runtime_path = os.path.join(tvm_source_dir, "apps", "microtvm", "zephyr", "aot_demo")
+    model_path = os.path.join(this_dir, "testdata", "ic_fp32.tflite")
+
+    # Import TFLite model
+    tflite_model_buf = open(model_path, "rb").read()
+    try:
+        import tflite
+
+        tflite_model = tflite.Model.GetRootAsModel(tflite_model_buf, 0)
+    except AttributeError:
+        import tflite.Model
+
+        tflite_model = tflite.Model.Model.GetRootAsModel(tflite_model_buf, 0)
+
+    # Load TFLite model and convert to Relay
+    relay_mod, params = relay.frontend.from_tflite(
+        tflite_model, shape_dict={"input_1": input_shape}, dtype_dict={"input_1 ": "float32"}
+    )
+
+    target = tvm.target.target.micro(model, options=["-link-params=1", "--executor=aot"])
+    with tvm.transform.PassContext(opt_level=3, config={"tir.disable_vectorize": True}):
+        lowered = relay.build(relay_mod, target, params=params)
+
+    # Load sample and generate input/output header files
+    sample = np.load(os.path.join(this_dir, "testdata", "ic_sample_fp32_8.npy"))
```

Comment:
311KB is quite a bit for this repo...i'm not sure whether we should load from a different place?

Reply:
replaced it with this link to its source:
[image classification model](https://github.com/eembc/ulpmark-ml/blob/fc1499c7cc83681a02820d5ddf5d97fe75d4f663/base_models/ic01/ic01_fp32.tflite)

Reply:
thanks, also possible to do so for this one?

Reply:
For the sample data I need to use a sample that has predictable result to test in the CI. I got this sample from a large dataset and need to keep it somewhere. The other option is to create a personal repository and add it there. We could reuse that repo later for other other samples as well. Let me know what you think.
Also I can save this numpy array as int8 and reduce the size 4 times if that works.